source/opt/aggressive_dead_code_elim_pass.cpp

```diff
@@ -218,12 +384,20 @@ bool AggressiveDCEPass::AggressiveDCE(ir::Function* func) {
     }
     worklist_.pop();
   }
-  // Mark all non-live instructions dead
-  for (auto& blk : *func) {
-    for (auto& inst : blk) {
-      if (live_insts_.find(&inst) != live_insts_.end())
+  // Mark all non-live instructions dead, except non-if branches
+  for (auto bi = structuredOrder.begin(); bi != structuredOrder.end(); ++bi) {
+    for (auto ii = (*bi)->begin(); ii != (*bi)->end(); ++ii) {
+      if (IsLive(&*ii))
         continue;
-      dead_insts_.insert(&inst);
+      if (IsBranch(ii->opcode())) {
```

Comment:
I was tempted to suggest use of IsStructuredIfHeader here.  Is this any better?

```
if (IsBranch(ii->opcode()) && IsStructuredIfHeader(*bi, nullptr, nullptr, nullptr)) continue;
```
But that would also need a tweak to IsStructuredHeader to allow null for those last three args.  You just want the bool result.

I don't know if that's worth it though.  Your choice.